aeon/io/reader.py

```diff
@@ -312,10 +312,22 @@ def __init__(self, pattern: str, model_root: str = "/ceph/aeon/aeon/data/process
     def read(self, file: Path) -> pd.DataFrame:
         """Reads data from the Harp-binarized tracking file."""
         # Get config file from `file`, then bodyparts from config file.
-        model_dir = Path(*Path(file.stem.replace("_", "/")).parent.parts[-4:])
-        config_file_dir = Path(self._model_root) / model_dir
-        if not config_file_dir.exists():
-            raise FileNotFoundError(f"Cannot find model dir {config_file_dir}")
+        model_dir = Path(*Path(file.stem.replace("_", "/")).parent.parts[1:])
```

Comment:
@jkbhagatio In the current data contract I don't see how this could change with locations. If anything counting backwards with a hard-coded folder count is even more likely to break if we ever change location of metadata folders.

The file format convention for all Aeon streams is the following:

`<DeviceName>/<DeviceName>_<Metadata>_<ChunkTime>.<Extension>`

So for example let's take the Harp encoder register, we might have:

`Patch1/Patch1_90_2024-03-02T12-00-00.bin`
- `DeviceName`: `Patch1`
- `Metadata`: `90` in this case refers to register 90
- `2024-03-02T12-00-00` is the chunk timestamp
- `.bin` is the extension

For a pose tracking file we might have the following:
`CameraTop/CameraTop_test-node1_topdown-multianimal-id-133_2024-03-02T12-00-00.bin`
- `DeviceName`: `CameraTop`
- `Metadata`: `test-node1_topdown-multianimal-id-133`
- `2024-03-02T12-00-00` is the chunk timestamp
- `.bin` is the extension

Now if we breakdown the logic I added there, you can see there are a few parts to it:

`Path(file.stem.replace("_", "/")).parent.parts[1:]`
- `file.stem`: this gets the filename without extension, so extension changes are irrelevant
- `.replace("_", "/")`: this turns all `_` into path separators. Our (possibly unwritten) convention is that all `_` characters are metadata separators, and there might be an arbitrary number of them.
- `Path.parent`: by convention all Aeon streams have the chunk timestamp as the last metadata attribute of a file, and this removes it from the dir path
- `.parts[1:]`: by convention all Aeon streams have the device name as the first metadata attribute of a stream file, and this removes it from the dir path.

My claim is that whatever remains is the model metadata ID in the case of SLEAP pose tracking. Is there any other assumption I missed?

Reply:
This is specific only to the pose data, and to the way we are writing out files.

Yeah, I guess as long as we never add any extra prefix between the device register and the training node / pc name, (between '202' and 'test-node1' for this example `CameraTop_202_test-node1_5571696_2024-08-29T00-00-00_topdown-multianimal-id_2024-09-21T23-00-00`), then this approach will always be equivalent with the previous?

Reply:
That's right, the exact metadata format is contingent on the specific file reader, but this is fine IMO.

You do raise a good point I might have inadvertently omitted the register metadata, I will double check.

The main goal here was to encode an approach where metadata components which are fixed are assumed fixed (i.e. device name, register address and chunk timestamp), but we leave variable components free to vary (i.e. the model path).

Reply:
ok cool sounds good